SYCL/DeviceLib/half_precision_math_test_marray_vec.cpp

```diff
@@ -0,0 +1,169 @@
+            buffer2, cgh);                                                     \
+        cgh.single_task<TypeHelper<class half_precision##NAME, T>>([=]() {     \
+          res_access[0] = sycl::half_precision::NAME(input_access[0]);         \
+        });                                                                    \
+      });                                                                      \
+    }                                                                          \
+    assert(checkEqual(result, ref));                                           \
+  }
+
+HALF_PRECISION_OPERATOR(sin)
+HALF_PRECISION_OPERATOR(tan)
+HALF_PRECISION_OPERATOR(cos)
+HALF_PRECISION_OPERATOR(exp)
+HALF_PRECISION_OPERATOR(exp2)
+HALF_PRECISION_OPERATOR(exp10)
+HALF_PRECISION_OPERATOR(log)
+HALF_PRECISION_OPERATOR(log2)
+HALF_PRECISION_OPERATOR(log10)
+HALF_PRECISION_OPERATOR(sqrt)
+HALF_PRECISION_OPERATOR(rsqrt)
+HALF_PRECISION_OPERATOR(recip)
+
+#undef HALF_PRECISION_OPERATOR
+
+#define HALF_PRECISION_OPERATOR_2(NAME)                                        \
+  template <typename T>                                                        \
+  void half_precision_math_test_2_##NAME(queue &deviceQueue, T result,         \
+                                         T input1, T input2, size_t ref) {     \
+    {                                                                          \
+      buffer<T, 1> buffer1(&result, 1);                                        \
+      buffer<T, 1> buffer2(&input1, 1);                                        \
+      buffer<T, 1> buffer3(&input2, 1);                                        \
+      deviceQueue.submit([&](handler &cgh) {                                   \
+        accessor<T, 1, access::mode::write, target::device> res_access(        \
+            buffer1, cgh);                                                     \
+        accessor<T, 1, access::mode::write, target::device> input1_access(     \
+            buffer2, cgh);                                                     \
+        accessor<T, 1, access::mode::write, target::device> input2_access(     \
+            buffer3, cgh);                                                     \
+        cgh.single_task<TypeHelper<class half_precision2##NAME, T>>([=]() {    \
+          res_access[0] =                                                      \
+              sycl::half_precision::NAME(input1_access[0], input2_access[0]);  \
+        });                                                                    \
+      });                                                                      \
+    }                                                                          \
+    assert(checkEqual(result, ref));                                           \
+  }
+
+HALF_PRECISION_OPERATOR_2(divide)
+HALF_PRECISION_OPERATOR_2(powr)
+
+#undef HALF_PRECISION_OPERATOR_2
+
+#define HALF_PRECISION_TESTS_3(TYPE)                                           \
```

Comment:
Could this be done with a template function instead? Should avoid the need for `COMMA`.